packages/warehouses/src/warehouseClients/BigqueryWarehouseClient.ts

```diff
@@ -215,9 +218,10 @@ export class BigqueryWarehouseClient extends WarehouseBaseClient<CreateBigqueryC
             });
 
             await streamPromise;
-        } catch (e) {
-            const response = e?.response as bigquery.IJob;
-            const responseError = response?.status?.errorResult || e;
+        } catch (e: unknown) {
+            const response = (e as { response?: bigquery.IJob })?.response;
+            const responseError =
+                response?.status?.errorResult || (e as bigquery.IErrorProto);
```

Comment:
this is still assuming with `as` and can be problematic when we update lib version
I suggest a small object checker that complies with IErrorProto

Reply:
yeah, I agree, we should not use `as` anywhere. I fixed this